core/src/main/java/com/netflix/msl/io/ThriftyUtf8Reader.java

```diff
@@ -32,430 +32,276 @@
  */
 public class ThriftyUtf8Reader extends Reader {
     /** Default byte buffer size (8192). */
-    public static final int DEFAULT_BUFFER_SIZE = 8192;
+    public static final int DEFAULT_BUFFER_SIZE = 16384;
```

Comment:
Any specific reason for doubling the buffer size?  Is this simply to account for possible malformed characters which could take up more space?

Reply:
A larger buffer is more efficient. I just picked 16384 as a reasonable balance but it could be even bigger.